File: Napakalaki Ruby/lib/card_dealer.rb
# encoding: UTF-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'treasure_kind'
require_relative 'bad_consequence'
require_relative 'prize'
require_relative 'treasure'
require_relative 'cultist'

module Napakalaki

  require "singleton"

  class CardDealer

    include Singleton

    attr_accessor :usedMonsters, :unusedMonsters, :usedTreasures, :unusedTreasures, :unusedCultist;

    def initTreasureCardDeck

      @unusedTreasures = Array.new
      @usedTreasures = Array.new

      treasure = Treasure.new("!Sí mi amo¡", 0, 4, 7, TreasureKind::HELMET)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Botas de investigación", 600, 3, 4, TreasureKind::SHOE)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Capucha de Cthulhu", 500, 3, 5, TreasureKind::HELMET)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("A prueba de babas", 400, 2, 5, TreasureKind::ARMOR)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Botas de lluvia ácida", 800, 1, 1, TreasureKind::SHOE)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Casco minero", 400, 2, 4, TreasureKind::HELMET)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Ametralladora Thompson", 600, 4, 8, TreasureKind::BOTHHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Camiseta de la UGR", 100, 1, 7, TreasureKind::ARMOR)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Clavo de rail ferroviario", 400, 3, 6, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Cuhillo de sushi arcano", 300, 2, 3, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Fez alópodo", 700, 3, 5, TreasureKind::HELMET)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Hacha prehistórica", 500, 2, 5, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("El aparato del Pr.Tesla", 900, 4, 8, TreasureKind::ARMOR)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Gaita", 500, 4, 5, TreasureKind::BOTHHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Insecticida", 300, 2, 3, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Escopeta de 3 cañones", 700, 4, 6, TreasureKind::BOTHHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Garabato místico", 300, 2, 2, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("La fuerza de Mr.T", 1000, 0, 0, TreasureKind::NECKLACE)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("La rebeca metálica", 400, 2, 3, TreasureKind::ARMOR)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Mazo de los antiguos", 200, 3, 4, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Necroplayboycón", 300, 3, 5, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Lanzallamas", 800, 4, 8, TreasureKind::BOTHHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Necrocomicón", 100, 1, 1, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Necronomicón", 800, 5, 7, TreasureKind::BOTHHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Linterna a 2 manos", 400, 3, 6, TreasureKind::BOTHHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Necrognomicón", 200, 2, 4, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Necrotelecom", 300, 2, 3, TreasureKind::HELMET)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Porra preternatural", 200, 2, 3, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Tentáculo de pega", 200, 0, 1, TreasureKind::HELMET)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Zapato deja-amigos", 500, 0, 1, TreasureKind::SHOE)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Shogulador", 600, 1, 1, TreasureKind::BOTHHAND)
      @unusedTreasures.push(treasure)

      treasure = Treasure.new("Varita de atizamiento", 400, 3, 4, TreasureKind::ONEHAND)
      @unusedTreasures.push(treasure)

    end


    def initMonsterCardDeck

      @unusedMonsters = Array.new
      @usedMonsters = Array.new

      prize = Prize.new(4,2) 
      badConsequence = BadConsequence.newLevelNumberOfTreasures('Pierdes 5 niveles y 3 tesoros visibles',5 , 3, 0)
      @unusedMonsters.push(Monster.newBadConsequencePrize('El rey de rosa',13,badConsequence,prize))

      prize = Prize.new(4,1) 
      badConsequence = BadConsequence.newLevelSpecificTreasures('Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y 1 mano oculta', 1, [TreasureKind::ONEHAND],[TreasureKind::ONEHAND])
      @unusedMonsters.push(Monster.newBadConsequencePrize('Ángeles de la noche ibicenca', 14, badConsequence, prize))

      prize = Prize.new(2,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Pierdes tu armadura visible y otra oculta', 1, [TreasureKind::ARMOR], [TreasureKind::ARMOR])
      @unusedMonsters.push(Monster.newBadConsequencePrize('3 Byakhees de bonanza', 8, badConsequence, prize))

      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Embobados con el lindo primigenio te descartas de tu casco visible', 1, [TreasureKind::HELMET], Array.new)
      @unusedMonsters.push(Monster.newBadConsequencePrize('Chibithulhu', 2, badConsequence, prize))

      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('El primordial bostezo contagioso. Pierdes el calzado visible', 1, [TreasureKind::SHOE], Array.new)
      @unusedMonsters.push(Monster.newBadConsequencePrize('El sopor de Dunwich', 2, badConsequence, prize))

      prize = Prize.new(3,1)
      badConsequence = BadConsequence.newLevelNumberOfTreasures('Pierdes todos tus tesoros visibles', 1, 10, 0)
      @unusedMonsters.push(Monster.newBadConsequencePrize('El gorrón en el umbral', 10, badConsequence, prize))

      prize = Prize.new(2,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Pierdes la armadura visible', 1, [TreasureKind::ARMOR], Array.new)
      @unusedMonsters.push(Monster.newBadConsequencePrize('H.P. Munchcraft', 6, badConsequence, prize))

      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Sientes bichos bajo la ropa. Descarta la armadura visible', 1, [TreasureKind::ARMOR], Array.new)
      @unusedMonsters.push(Monster.newBadConsequencePrize('Bichgooth', 2, badConsequence, prize))

      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelNumberOfTreasures('Toses los pulmones y pierdes 2 niveles', 2, 0, 0)
      @unusedMonsters.push(Monster.newBadConsequencePrize('La que redacta en las tinieblas', 10, badConsequence, prize))

      prize = Prize.new(2,1)
      badConsequence = BadConsequence.newDeath('Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estas muerto')
      @unusedMonsters.push(Monster.newBadConsequencePrize('Los hondos', 8, badConsequence, prize))

      prize = Prize.new(2,1)
      badConsequence = BadConsequence.newLevelNumberOfTreasures('Pierdes 2 niveles y 2 tesoros ocultos', 2, 0, 2)
      @unusedMonsters.push(Monster.newBadConsequencePrize('Semillas Cthulhu', 4, badConsequence, prize))

      prize = Prize.new(2,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Te intentas escaquear.Pierdes una mano visible', 1, [TreasureKind::ONEHAND], Array.new)
      @unusedMonsters.push(Monster.newBadConsequencePrize('Dameargo', 1, badConsequence, prize))

      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelNumberOfTreasures('Da mucho asquito.Pierdes 3 niveles', 3, 0, 0)
      @unusedMonsters.push(Monster.newBadConsequencePrize('Pollipólipo volante', 3, badConsequence, prize))

      prize = Prize.new(3,1)
      badConsequence = BadConsequence.newDeath('No le hace gracia que pronuncien mal su nombre. Estas muerto')
      @unusedMonsters.push(Monster.newBadConsequencePrize('Yskhtihyssg-Goth', 12, badConsequence, prize))

      prize = Prize.new(4,1)
      badConsequence = BadConsequence.newDeath('La familia te atrapa. Estas muerto')
      @unusedMonsters.push(Monster.newBadConsequencePrize('Familia feliz', 1, badConsequence, prize))

      prize = Prize.new(2,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('La quinta directiva primaria te obliga a perder 2 niveles y 2 manos visible', 2, [TreasureKind::ONEHAND,TreasureKind::ONEHAND,TreasureKind::BOTHHAND], Array.new)
      @unusedMonsters.push(Monster.newBadConsequencePrize('Roboggoth', 8, badConsequence, prize))

      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Te asusta en la noche. Pierdes un casco visible', 1, [TreasureKind::HELMET], Array.new)
      @unusedMonsters.push(Monster.newBadConsequencePrize('El espia', 5, badConsequence, prize))

      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelNumberOfTreasures('Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles', 2, 5, 0)
      @unusedMonsters.push(Monster.newBadConsequencePrize('El Lenguas', 20, badConsequence, prize))

      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos', 1, [TreasureKind::ONEHAND, TreasureKind::ONEHAND, TreasureKind::BOTHHAND], Array.new)
      @unusedMonsters.push(Monster.newBadConsequencePrize('Bicéfalo', 20, badConsequence, prize))
      
      prize = Prize.new(3,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Pierdes una mano visible', 1, [TreasureKind::ONEHAND], Array.new)
      @unusedMonsters.push(Monster.newBadConsequenceCultist('El mal indecible impronunciable', 10, badConsequence, prize, -2))
      
      prize = Prize.new(2,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Pierdes tus tesoros visible. Jajaja', 1, [TreasureKind::ONEHAND, TreasureKind::ONEHAND, TreasureKind::BOTHHAND, TreasureKind::HELMET, TreasureKind::ARMOR, TreasureKind::SHOE, TreasureKind::NECKLACE], Array.new)
      @unusedMonsters.push(Monster.newBadConsequenceCultist('Testigos Oculares', 6, badConsequence, prize, +2))
      
      prize = Prize.new(2,5)
      badConsequence = BadConsequence.newDeath('Hoy no es tu dia de suerte.Mueres')
      @unusedMonsters.push(Monster.newBadConsequenceCultist('El gran cthulhu', 20, badConsequence, prize, +4))
      
      prize = Prize.new(2,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Tu gobierno te recorta 2 niveles', 2, Array.new, Array.new)
      @unusedMonsters.push(Monster.newBadConsequenceCultist('Serpiente Político', 8, badConsequence, prize, -2))
      
      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Pierdes tu casco y tu armadura visible. Pierdes tus manos ocultas', 1, [TreasureKind::HELMET, TreasureKind::ARMOR], [TreasureKind::ONEHAND, TreasureKind::ONEHAND, TreasureKind::BOTHHAND])
      @unusedMonsters.push(Monster.newBadConsequenceCultist('Felpuggoth', 2, badConsequence, prize, +5))
      
      prize = Prize.new(4,2)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Pierdes 2 niveles', 2, Array.new, Array.new)
      @unusedMonsters.push(Monster.newBadConsequenceCultist('Shoggoth', 16, badConsequence, prize, -4))
      
      prize = Prize.new(1,1)
      badConsequence = BadConsequence.newLevelSpecificTreasures('Pintalabios negro. Pierdes 2 niveles', 2, Array.new, Array.new)
      @unusedMonsters.push(Monster.newBadConsequenceCultist('Lolitagooth', 2, badConsequence, prize, +3))
      
    end
    
    def initCultistCardDeck

      @unusedCultists = Array.new
      
      cultist = Cultist.new("Sectario", 1)
      @unusedCultists.push(cultist)
        
      cultist = Cultist.new("Sectario", 2)
      @unusedCultists.push(cultist)

      cultist = Cultist.new("Sectario", 1)
      @unusedCultists.push(cultist)

      cultist = Cultist.new("Sectario", 2)
      @unusedCultists.push(cultist)

      cultist = Cultist.new("Sectario", 1)
      @unusedCultists.push(cultist)

      cultist = Cultist.new("Sectario", 1)
      @unusedCultists.push(cultist)
      
      
    end

    def shuffleTreasures

      @unusedTreasures.shuffle!

    end


    def shuffleMonsters

      @unusedMonsters.shuffle!

    end
    
    def shuffleCultist
      
      @unusedCultists.shuffle!
      
    end

    def nextTreasure

      if @unusedTreasures.empty? then

        while !@usedTreasures.empty? do

          @unusedTreasures.push(@usedTreasures[0])

          @usedTreasures.delete_at(0)

        end

        shuffleTreasures

      end

      nextTrs = @unusedTreasures[0]

      @unusedTreasures.delete_at(0)

      return nextTrs

    end


    def nextMonster

      if @unusedMonsters.empty? then

        while !@usedMonsters.empty? do

          @unusedMonsters.push(@usedMonsters[0])

          @usedMonsters.delete_at(0)

        end

        shuffleMonsters

      end

      nextMnst = @unusedMonsters[0]

      @unusedMonsters.delete_at(0)

      return nextMnst

    end
    
    def nextCultist
      
      nextCultist = @unusedCultists[0]
      
      @unusedCultists.delete_at(0)
      
      return nextCultist
      
    end


    def giveTreasureBack (treasure)

      @usedTreasures.push(treasure)    

    end


    def giveMonsterBack (monster)

      @usedMonsters.push(monster)    

    end


    def initCards

      initTreasureCardDeck
      shuffleTreasures
      initMonsterCardDeck
      shuffleMonsters
      initCultistCardDeck
      shuffleCultist

    end

  end
  
end

File: Napakalaki Java/src/NapakalakiGame/CardDealer.java
/*
 */
package NapakalakiGame;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;


public class CardDealer {
    
    private static final CardDealer instance = new CardDealer();
    
    private ArrayList<Treasure> unusedTreasures = new ArrayList();
    private ArrayList<Treasure> usedTreasures = new ArrayList();
    private ArrayList<Monster> unusedMonsters = new ArrayList();
    private ArrayList<Monster> usedMonsters = new ArrayList();
    private ArrayList<Cultist> unusedCultists = new ArrayList();

    
    private CardDealer() {
    }
    
    public static CardDealer getInstance() {
        return instance;
    }
    
    private void initTreasureCardDeck(){

        Treasure treasure = new Treasure("¡Sí, mi amo!", 0, 4 ,7, TreasureKind.helmet);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Botas de investigación", 600, 3 ,4, TreasureKind.shoe);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Capucha de Cthulhu", 500, 3, 5, TreasureKind.helmet);
        unusedTreasures.add(treasure);

        treasure = new Treasure("A prueba de babas", 400, 2, 5, TreasureKind.armor);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Botas de lluvia ácida", 800, 1, 1, TreasureKind.shoe);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Casco minero", 400, 2, 4, TreasureKind.helmet);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Ametralladora Thompson", 600, 4, 8, TreasureKind.bothHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Camiseta de la UGR", 100, 1, 7, TreasureKind.armor);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Clavo de raíl ferroviario", 400, 3, 6, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Cuchillo de sushi arcano", 300, 2, 3, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Fez alópodo", 700, 3, 5, TreasureKind.helmet);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Hacha prehistórica", 500, 2, 5, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("El aparato del Pr. Tesla", 900, 4, 8, TreasureKind.armor);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Gaita", 500, 4, 5, TreasureKind.bothHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Insecticida", 300, 2, 3, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Escopeta de 3 cañones", 700, 4, 6, TreasureKind.bothHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Garabato místico", 300, 2, 2, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("La fuerza de Mr.T", 1000, 0, 0, TreasureKind.necklace);
        unusedTreasures.add(treasure);

        treasure = new Treasure("La rebeca metálica", 400, 2, 3, TreasureKind.armor);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Mazo de los antiguos", 200, 3, 4, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Necroplayboycón", 300, 3, 5, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Lanzallamas", 800, 4, 8, TreasureKind.bothHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Necrocomicón", 100, 1, 1, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Necronomicón", 800, 5, 7, TreasureKind.bothHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Linterna a 2 manos", 400, 3, 6, TreasureKind.bothHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Necrognomicón", 200, 2, 4, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Necrotelecom", 300, 2, 3, TreasureKind.helmet);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Porra preternatural", 200, 2, 3, TreasureKind.oneHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Tentáculo de pega", 200, 0, 1, TreasureKind.helmet);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Zapato deja-amigos", 500, 0, 1, TreasureKind.shoe);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Shogulador", 600, 1, 1, TreasureKind.bothHand);
        unusedTreasures.add(treasure);

        treasure = new Treasure("Varita de atizamiento", 400, 3, 4, TreasureKind.oneHand);
        unusedTreasures.add(treasure);
        
    }
    
    private void initMonsterCardDeck(){
        
        BadConsequence badConsequence = new BadConsequence("Pierdes 5 niveles y 3 tesoros visibles.", 5, 3, 0);
        Prize prize = new Prize(4,2);
        unusedMonsters.add(new Monster("El rey de rosa", 13, badConsequence, prize));
        
        badConsequence = new BadConsequence("Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y 1 mano oculta.",
                1, new ArrayList(Arrays.asList(TreasureKind.oneHand)), new ArrayList(Arrays.asList(TreasureKind.oneHand)));
        prize = new Prize(4,1);
        unusedMonsters.add(new Monster("Ángeles de la noche ibicenca", 14, badConsequence, prize));
        
        badConsequence = new BadConsequence("Pierdes tu armadura visible y otra oculta.", 1, new ArrayList(Arrays.asList(TreasureKind.armor)),new ArrayList(Arrays.asList(TreasureKind.armor)));
        prize = new Prize(2,1);
        unusedMonsters.add(new Monster("3 Byakhees de bonanza", 8, badConsequence, prize));
        
        badConsequence = new BadConsequence("Embobado con el lindo primigenio, te descartas de tu casco visible.", 1, new ArrayList(Arrays.asList(TreasureKind.helmet)),new ArrayList());
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("Chibithulhu", 2, badConsequence, prize));
        
        badConsequence = new BadConsequence("El primordial bostezo contagioso. Pierdes el calzado visible.", 1, new ArrayList(Arrays.asList(TreasureKind.shoe)), new ArrayList());
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("El sopor de Dunwich", 2, badConsequence, prize));
        
        badConsequence = new BadConsequence("Pierdes todos tus tesoros visibles.", 1, 10, 0);
        prize = new Prize(3,1);
        unusedMonsters.add(new Monster("El gorrón en el umbral", 10, badConsequence, prize));
        
        badConsequence = new BadConsequence("Pierdes la armadura visible.", 1, new ArrayList(Arrays.asList(TreasureKind.armor)), new ArrayList());
        prize = new Prize(2,1);
        unusedMonsters.add(new Monster("H.P. Munchcraft", 6, badConsequence, prize));
        
        badConsequence = new BadConsequence("Sientes bichos bajo la ropa. Descarta la armadura visible.", 1, new ArrayList(Arrays.asList(TreasureKind.armor)), new ArrayList());      
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("Bichgooth", 2, badConsequence, prize));
        
        badConsequence = new BadConsequence("Toses los pulmones y pierdes 2 niveles.", 2, 0, 0);
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("La que redacta en las tinieblas", 2, badConsequence, prize));
        
        badConsequence = new BadConsequence("Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estás muerto.", true);
        prize = new Prize(2,1);
        unusedMonsters.add(new Monster("Los hondos", 8, badConsequence, prize));
        
        badConsequence = new BadConsequence("Pierde 2 niveles y 2 tesoros ocultos.", 2, 0, 2);
        prize = new Prize(2,1);
        unusedMonsters.add(new Monster("Semillas Cthulhu", 4, badConsequence, prize));
        
        badConsequence = new BadConsequence("Te intentas escaquear. Pierdes una mano visible.", 1, new ArrayList(Arrays.asList(TreasureKind.oneHand)), new ArrayList());
        prize = new Prize(2,1);
        unusedMonsters.add(new Monster("Dameargo", 1, badConsequence, prize));
        
        badConsequence = new BadConsequence("Da mucho asquito. Pierdes 3 niveles.", 3, 0, 0);
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("Pollipólipo volante", 3, badConsequence, prize));
        
        badConsequence = new BadConsequence("No le hace gracia que pronuncien mal su nombre. Estás muerto.", true);
        prize = new Prize(3,1);
        unusedMonsters.add(new Monster("Yskhtihyssg-Goth", 12, badConsequence, prize));
        
        badConsequence = new BadConsequence("La familia te atrapa. Estás muerto.", true);
        prize = new Prize(4,1);
        unusedMonsters.add(new Monster("Familia feliz", 1, badConsequence, prize));
        
        badConsequence = new BadConsequence("La quinta directiva primaria te obliga a perder 2 niveles y un tesoro 2 manos visible.", 2, new ArrayList(Arrays.asList(TreasureKind.bothHand)), new ArrayList());
        prize = new Prize(2,1);
        unusedMonsters.add(new Monster("Roboggoth", 8, badConsequence, prize));
        
        badConsequence = new BadConsequence("Te asusta en la noche. Pierdes un casco visible.", 1, new ArrayList(Arrays.asList(TreasureKind.helmet)), new ArrayList());
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("El espía", 5, badConsequence, prize));
        
        badConsequence = new BadConsequence("Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles.", 2, 5, 0);
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("El Lenguas", 20, badConsequence, prize));
        
        badConsequence = new BadConsequence("Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos.", 3, new ArrayList(Arrays.asList(TreasureKind.oneHand,TreasureKind.oneHand,TreasureKind.bothHand)), new ArrayList());
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("Bicéfalo", 20, badConsequence, prize));
        
        badConsequence = new BadConsequence("Pierdes 1 mano visible.", 1, new ArrayList(Arrays.asList(TreasureKind.oneHand)), new ArrayList());
        prize = new Prize(3,1);
        unusedMonsters.add(new Monster("El mal indecible impronunciable", 10, badConsequence, prize, -2));
        
        badConsequence = new BadConsequence("Pierdes tus tesoros visibles. Jajaja.", 1, 10, 0);
        prize = new Prize(2,1);
        unusedMonsters.add(new Monster("Testigos oculares", 6, badConsequence, prize, 2));
        
        badConsequence = new BadConsequence("Hoy no es tu día de suerte. Mueres.", true);
        prize = new Prize(2,5);
        unusedMonsters.add(new Monster("El gran cthulhu", 20, badConsequence, prize, 4));
        
        badConsequence = new BadConsequence("Tu gobierno te recorta 2 niveles.", 2, 0, 0);
        prize = new Prize(2,1);
        unusedMonsters.add(new Monster("Serpiente Político", 8, badConsequence, prize, -2));
        
        badConsequence = new BadConsequence("Pierdes tu casco y tu armadura visible. Pierdes tus manos ocultas.", 1, new ArrayList(Arrays.asList(TreasureKind.helmet, TreasureKind.armor)), new ArrayList(Arrays.asList(TreasureKind.bothHand, TreasureKind.oneHand, TreasureKind.oneHand)));
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("Felpuggoth", 2, badConsequence, prize, 5));
        
        badConsequence = new BadConsequence("Pierdes 2 niveles.", 2, 0, 0);
        prize = new Prize(4,2);
        unusedMonsters.add(new Monster("Shoggoth", 16, badConsequence, prize, -4));
        
        badConsequence = new BadConsequence("Pintalabios negro. Pierdes 2 niveles.", 2, 0, 0);
        prize = new Prize(1,1);
        unusedMonsters.add(new Monster("Lolitagooth", 2, badConsequence, prize, 3));
        
    }
    
    private void initCultistsCardDeck(){
        
        Cultist cultist = new Cultist("Sectario", 1);
        unusedCultists.add(cultist);
        
        cultist = new Cultist("Sectario", 2);
        unusedCultists.add(cultist);
        
        cultist = new Cultist("Sectario", 1);
        unusedCultists.add(cultist);
        
        cultist = new Cultist("Sectario", 2);
        unusedCultists.add(cultist);
        
        cultist = new Cultist("Sectario", 1);
        unusedCultists.add(cultist);
        
        cultist = new Cultist("Sectario", 1);
        unusedCultists.add(cultist);
        
    }
    
    private void shuffleTreasures(){
    
         Collections.shuffle(unusedTreasures);
        
    }
    
    private void shuffleMonsters(){
    
        Collections.shuffle(unusedMonsters);
        
    }
    
    private void shuffleCultists(){
    
        Collections.shuffle(unusedCultists);
        
    }
    
    public Treasure nextTreasure(){
        
        if (unusedTreasures.isEmpty()){
            
            while (!usedTreasures.isEmpty()){
                
                unusedTreasures.add(usedTreasures.get(0));
                
                usedTreasures.remove(0);
                
            }
            
            shuffleTreasures();
            
        }
        
        Treasure nextTreasure = unusedTreasures.get(0);
        
        unusedTreasures.remove(0);
        
        return nextTreasure;
        
    }
    
    public Monster nextMonster(){
    
        if (unusedMonsters.isEmpty()){
            
            while (!usedMonsters.isEmpty()){
                
                unusedMonsters.add(usedMonsters.get(0));
                
                usedMonsters.remove(0);
                
            }
            
            shuffleMonsters();
            
        }
        
        Monster nextMonster = unusedMonsters.get(0);
        
        unusedMonsters.remove(0);
        
        return nextMonster;
        
    }
    
    public Cultist nextCultist(){
        
        Cultist nextCultist = unusedCultists.get(0);
        
        unusedCultists.remove(0);
        
        return nextCultist;
        
    }
    
    public void giveTreasureBack(Treasure t){
    
        usedTreasures.add(t);
        
    }
    
    public void giveMonsterBack(Monster m){
        
        usedMonsters.add(m);
        
    }
    
    public void initCards(){
    
        initTreasureCardDeck();
        shuffleTreasures();
        initMonsterCardDeck();
        shuffleMonsters();
        initCultistsCardDeck();
        shuffleCultists();
    }
    
}
